pmm/summary.go

```diff
@@ -0,0 +1,263 @@
+/*
+	Copyright (c) 2016, Percona LLC and/or its affiliates. All rights reserved.
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>
+*/
+
+package pmm
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+	"regexp"
+	"strings"
+	"io"
+	"archive/zip"
+	"path/filepath"
+)
+
+// Collector parameters and description.
+type Collector struct {
+	CollectorDescription	string
+	ExecCommand		[]string
+	OutputFileName		string
+}
+
+// CollectData runs a command and collects output into a file.
+func (c Collector) CollectData() {
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#receiver-type
I think the last case should be applied there:
> Finally, when in doubt, use a pointer receiver.